### PR TITLE
Prodcon ubench xdt support

### PR DESCRIPTION
**Pull request by shyamjesal**
- Adds transfer via XDT as an option.
- create separate knative yamls per transfer type
  - add env variable TRANSFER_TYPE to knative yamls 
- add S3 support

XDT support requires pulling private go modules. Private go modules are fetched during the docker build process. `--secret` command in `DOCKER_BUILDKIT` is used to pass access keys to the private repo. Ref. [this](https://pythonspeed.com/articles/docker-build-secrets/) for more info.

### CODE REVIEW

File: function-images/tests/chained-function-serving/consumer/consumer.go
Comment (ustiugov) [inline function-images/tests/chained-function-serving/consumer/consumer.go:41]: need to ensure that people outside of the vHive core team are able to use this benchmark. please check if the build succeeds, otherwise need to find another trick

File: function-images/tests/chained-function-serving/producer/producer.go
Comment (ustiugov) [inline function-images/tests/chained-function-serving/producer/producer.go:61]: avoid informal messages

**Review by ustiugov — COMMENTED** [review]
please get a review from Michal too

File: function-images/tests/chained-function-serving/consumer/consumer.go
Comment (shyamjesal) [reply function-images/tests/chained-function-serving/consumer/consumer.go:41]: need to find another trick, build will not succeed since it needs private repo access. 
We will have to create separate repo for SDK.

**Review by MBaczun — APPROVED** [review]

File: function-images/tests/chained-function-serving/producer/producer.go
Comment (ustiugov) [inline function-images/tests/chained-function-serving/producer/producer.go:115]: why not 4096? that's weird

**Review by ustiugov — COMMENTED** [review]
I don't get how it could work for an external user.

File: function-images/tests/chained-function-serving/consumer/consumer.go
Comment (ustiugov) [inline function-images/tests/chained-function-serving/consumer/consumer.go:94]: Please handle invalid values

File: function-images/tests/chained-function-serving/consumer/consumer.go
Comment (ustiugov) [inline function-images/tests/chained-function-serving/consumer/consumer.go:101]: Redundant

File: function-images/tests/chained-function-serving/producer/producer.go
Comment (ustiugov) [inline function-images/tests/chained-function-serving/producer/producer.go:148]: Handle invalid values

**Review by ustiugov — CHANGES_REQUESTED** [review]

File: function-images/tests/chained-function-serving/producer/producer.go
Comment (shyamjesal) [reply function-images/tests/chained-function-serving/producer/producer.go:115]: it exceeds the gRPC limit by a few bytes.

**Review by ustiugov — COMMENTED** [review]
please add a bash script in `function-images/tests` that would run `envsubst` and generate the manifests with AWS credentials for kubernetes deployment

**Comment by ustiugov** [comment]
stale branch, please rebase

**Comment by ustiugov** [comment]
actually, let's wait with rebasing, till Bora's PR is merged...

File: function-images/tests/chained-function-serving/Dockerfile
Comment (ustiugov) [inline function-images/tests/chained-function-serving/Dockerfile:50]: why is it necessary?

**Review by ustiugov — COMMENTED** [review]
lgtm

File: function-images/tests/chained-function-serving/Dockerfile
Comment (shyamjesal) [reply function-images/tests/chained-function-serving/Dockerfile:50]: S3 operations need the cs-certificates.

**Review by ustiugov — DISMISSED** [review]

**Comment by ustiugov** [comment]
ok, now carefully rebase please

**Review by ustiugov — APPROVED** [review]

**Comment by ustiugov** [comment]
Please fix the broken links

**Review by ustiugov — APPROVED** [review]

**Pull request merged**
